crates/telio-lana/src/lib.rs

```diff
@@ -115,3 +135,196 @@ pub fn deinit_lana() -> Result<moose::Result, moose::Error> {
 pub fn is_lana_initialized() -> bool {
     MOOSE_INITIALIZED.load(DEFAULT_ORDERING)
 }
+
+/// Initialize different moose variations (real, file or mock)
+///
+/// # Parameters:
+/// * event_path - path of the DB file where events will be stored. If such file does not exist, it will be created, otherwise reused.
+/// * app_version - Indicates the semantic version of the application.
+/// * prod - whether the events should be sent to production or not
+/// * tx - channel tx half for the InitCallback instance
+pub fn init_moose(
+    event_path: String,
+    app_version: String,
+    prod: bool,
+    tx: SyncSender<bool>,
+) -> std::result::Result<moose::Result, moose::Error> {
+    cfg_if! {
+        if #[cfg(test)] {
+            test::MOCK.with(|m| {
+                let mock = m.borrow();
+                if mock.is_some() {
+                    mock.as_ref().unwrap().init(
+                        event_path.clone(),
+                        LANA_APP_NAME.to_string(),
+                        app_version.clone(),
+                        LANA_MOOSE_VERSION.to_string(),
+                        prod,
+                        Box::new(moose_callbacks::MooseInitCallback {
+                            init_success_tx: tx,
+                        }),
+                        Box::new(moose_callbacks::MooseErrorCallback),
+                    )
+                } else {
+                    moose::init(
+                        event_path,
+                        LANA_APP_NAME.to_string(),
+                        app_version,
+                        LANA_MOOSE_VERSION.to_string(),
+                        prod,
+                        Box::new(moose_callbacks::MooseInitCallback {
+                            init_success_tx: tx,
+                        }),
+                        Box::new(moose_callbacks::MooseErrorCallback),
+                    )
+                }
+            })
+        } else {
+            moose::init(
+                event_path,
+                LANA_APP_NAME.to_string(),
+                app_version,
+                LANA_MOOSE_VERSION.to_string(),
+                prod,
+                Box::new(moose_callbacks::MooseInitCallback {
+                    init_success_tx: tx,
+                }),
+                Box::new(moose_callbacks::MooseErrorCallback),
+            )
+        }
+    }
+}
+#[cfg(test)]
+mod test {
+    use super::event_log::moose;
+    use super::moose::{ErrorCallback, InitCallback};
+    use super::RefCell;
+    use super::{deinit_lana, init_lana, is_lana_initialized};
+
+    thread_local!(pub static MOCK: RefCell<Option<MooseMock>> = RefCell::new(None));
+
+    pub struct MooseMock {
```

Comment:
nit: Since we are not using any mocking library I think I would go with a `MooseStub` to make it more clear to readers not to expect eg. `mockall`.

Reply:
Changed.